test/helpers/helpers.go

```diff
@@ -0,0 +1,130 @@
+	// by the CreateOperatorSource function.
+	TestOperatorSourceLabelKey string = "opsrc-group"
+
+	// TestOperatorSourceLabelValue is a label value added to the opeator source returned
+	// by the CreateOperatorSource function.
+	TestOperatorSourceLabelValue string = "Community"
+)
+
+// WaitForResult polls the cluster for a particular resource name and namespace.
+// If the request fails because of an IsNotFound error it retries until the specified timeout.
+// If it succeeds it sets the result runtime.Object to the requested object.
+func WaitForResult(t *testing.T, result runtime.Object, namespace, name string) error {
+	// Get global framework variables
+	f := test.Global
+
+	namespacedName := types.NamespacedName{Name: name, Namespace: namespace}
+	err := wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = f.Client.Get(context.TODO(), namespacedName, result)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				t.Logf("Waiting for creation of %s runtime object\n", name)
+				return false, nil
+			}
+			return false, err
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+	t.Logf("Runtime object %s has been created\n", name)
+	return nil
+}
+
+// WaitForSuccessfulDeployment checks if a given deployment has readied all of
+// its replicas. If it has not, it retries until the deployment is ready or it
+// reaches the timeout.
+func WaitForSuccessfulDeployment(t *testing.T, deployment apps.Deployment) error {
+	// Get global framework variables
+	f := test.Global
+
+	// If deployment is already ready, lets just return.
+	if deployment.Status.ReadyReplicas == *deployment.Spec.Replicas {
+		return nil
+	}
+
+	namespacedName := types.NamespacedName{Name: deployment.Name, Namespace: deployment.Namespace}
+	result := &apps.Deployment{}
+	err := wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = f.Client.Get(context.TODO(), namespacedName, result)
+		if err != nil {
+			return false, err
+		}
+		if *deployment.Spec.Replicas == result.Status.ReadyReplicas {
+			return true, nil
+		}
+		t.Logf("Waiting for deployment %s to have (%d/%d) replicas ready\n", deployment.Name, result.Status.ReadyReplicas,
+			*deployment.Spec.Replicas)
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+	t.Logf("Deployment %s has been initialized successfully\n", deployment.Name)
+	return nil
+}
+
+// CreateRuntimeObject creates a runtime object using the test framework.
+func CreateRuntimeObject(f *test.Framework, ctx *test.TestCtx, obj runtime.Object) error {
+	return f.Client.Create(
+		context.TODO(),
+		obj,
+		&test.CleanupOptions{
+			TestContext:   ctx,
+			Timeout:       time.Second * 30,
+			RetryInterval: time.Second * 1,
+		})
+}
+
+// CreateOperatorSource returns an OperatorSource that can be used by tests that depend on an existing OperatorSource.
+func CreateOperatorSource(namespace string) *marketplace.OperatorSource {
```

Comment:
We should rename this to `CreateOperatorSourceDefinition` or something along those lines as we are not actually creating the object here.

Reply:
Resolved in the latest commit.